RecoEgamma/EgammaTools/plugins/EGEnergyAnalyzer.cc

```diff
@@ -53,15 +55,18 @@ class EGEnergyAnalyzer : public edm::one::EDAnalyzer<> {
   void analyze(const edm::Event&, const edm::EventSetup&) override;
   void endJob() override;
 
+  EGEnergyCorrectorFactoryFromEventSetup corddbFactory_;
   EGEnergyCorrector corfile;
-  EGEnergyCorrector cordb;
+  std::optional<EGEnergyCorrector> cordb;
 
   edm::EDGetTokenT<EcalRecHitCollection> ebRHToken_, eeRHToken_;
   const EcalClusterLazyTools::ESGetTokens ecalClusterToolsESGetTokens_;
 };
 
 EGEnergyAnalyzer::EGEnergyAnalyzer(const edm::ParameterSet& iConfig)
-    : ecalClusterToolsESGetTokens_{consumesCollector()} {
+    : corddbFactory_(consumesCollector(), "wgbrph"),
+      corfile(egEnergyCorrectorFactoryFromRootFile("/afs/cern.ch/user/b/bendavid/cmspublic/gbrv3ph.root")),
```

Comment:
by the look of this default parameter, does this file rather belong in /test/? Or maybe it's not needed any more at all?

Reply:
As I am not the maintainer of the code, I'd prefer to have any such work done in a later PR.